File: devops-scm-spring-boot-starter/src/main/java/com/tencent/devops/scm/spring/manager/ScmConnectorFactory.java
package com.tencent.devops.scm.spring.manager;

import com.tencent.devops.scm.sdk.common.connector.ScmConnector;
import com.tencent.devops.scm.spring.properties.HttpClientProperties;

public interface ScmConnectorFactory {

    ScmConnector create(HttpClientProperties properties);
}

File: devops-scm-spring-boot-starter/src/main/kotlin/com/tencent/devops/scm/spring/manager/ScmConnectorFactory.kt
package com.tencent.devops.scm.spring.manager

import com.tencent.devops.scm.sdk.common.connector.ScmConnector
import com.tencent.devops.scm.spring.properties.HttpClientProperties

interface ScmConnectorFactory {
    /**
     * 创建SCM连接器
     *
     * @param properties HTTP客户端配置属性
     * @return SCM连接器实例
     */
    fun create(properties: HttpClientProperties): ScmConnector
}